extensions/azurecore/src/utils.ts

```diff
@@ -161,6 +166,118 @@ export async function updateTenantIgnoreList(tenantIgnoreList: string[]): Promis
 	await configuration.update(constants.Filter, tenantIgnoreList, vscode.ConfigurationTarget.Global);
 }
 
+export function updateProviderSettings(defaultSettings: ProviderSettings[]): ProviderSettings[] {
+	let providerSettingsJson: ProviderSettingsJson[] | undefined = vscode.workspace.getConfiguration(constants.AzureSection).get(constants.ProviderSettingsJson) as ProviderSettingsJson[];
+	vscode.workspace.onDidChangeConfiguration(async (changeEvent) => {
+		const impactProvider = changeEvent.affectsConfiguration(constants.ProviderSettingsJsonSection);
+		if (impactProvider === true) {
+			await displayReloadAds(constants.ProviderSettingsJsonSection);
+		}
+	});
+	if (providerSettingsJson && providerSettingsJson.length > 0) {
+		try {
+			for (let cloudProvider of providerSettingsJson) {
+				// build provider setting
+				let newSettings = buildProviderSettings(cloudProvider);
+				defaultSettings.push(newSettings)
+				Logger.info(`Custom provider settings loaded for ${cloudProvider.settings.metadata.displayName}`);
+			}
+			void vscode.window.showInformationMessage(localize('providerSettings.success', 'Successfully loaded custom endpoints from settings'));
+
+		} catch (error) {
+			void vscode.window.showErrorMessage(localize('providerSettings.error', 'Could not load endpoints from settings, please check the logs for more details.'));
+			console.error(error.message);
+			throw Error(error.message);
+		}
+	}
+	return defaultSettings;
+}
+
+function buildProviderSettings(cloudProvider: ProviderSettingsJson): ProviderSettings {
```

Comment:
Consider updating the names for this stuff to make it clear it's for custom cloud providers - e.g. this would be `buildCustomCloudProviderSettings`